libraries/src/Application/ConsoleApplication.php

```
@@ -539,6 +545,45 @@ protected function populateHttpHost()
         $_SERVER['HTTPS']       = $uri->getScheme() === 'https' ? 'on' : 'off';
     }
 
+    /**
+     * Populates the user when a console option is set.
+     *
+     * @return  void
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function populateUser(): void
+    {
+        // Load the user when specified
+        $userOption = $this->getConsoleInput()->getParameterOption(['--user'], null);
+
+        if ($userOption === null) {
+            return;
+        }
+
+        try {
+            $factory = $this->getUserFactory();
+        } catch (\UnexpectedValueException $e) {
+            return;
```

Why do you hide the exception here? If the `--user` option is set and the application can't populate the specified user, it should show an error an exit.

---

This is a good question what should happen then.

---

But this exception is about the issue when no user factory is set in the application to load the user.

---

Maybe we could just throw another exception:
```php
throw new \Exception('User ' . $userOption . ' cannot be populated');
```
But, I don't understand why `try/catch` is needed here at all.

---

This catches the exception when the user factory dependency is not set. It ensures backwards compatibility.